Test Packages (#9896)
sorbet/sorbet: website/docs/error-reference.md

- the `__package.rb` is missing `export`s
- the `__package.rb` is missing `visible_to`, and `--gen-packages-update-visibility-for` flag is passed.

## 3735

> This error is specific to Stripe's custom `--sorbet-packages` mode. If you are at Stripe, please see [go/modularity](http://go/modularity) for more.

> This error occurs in `--experimental-test-packages` mode, when an invalid import was detected. There are two different scenarios that will produce this error, let's consider the following set of packages:

Reviewer: ```suggestion
This error occurs when an invalid import was detected. There are two different scenarios that will produce this error, let's consider the following set of packages:
```

I'd suggest making this early, unless you think you'll remember to come edit this after eventually removing the `--experimental-test-packages` flag.

> This error occurs in `--experimental-test-packages` mode, when an invalid import was detected. There are two different scenarios that will produce this error, let's consider the following set of packages:

Reviewer: ```suggestion
This error occurs in `--experimental-test-packages` mode, when an invalid import was detected. There are two different scenarios that will produce this error; let's consider the following set of packages:
```

or `.`

> - `A` - An application package that imports one constant `A::C1` and keeps `A::C2` private

Reviewer: ```suggestion
- `A` - An application package that exports one constant `A::Pub` and keeps `A::Priv` private
```

I suggested the different names so that people wouldn't have to remember which was which when they were later referenced, though I'm not sure whether these names matter, because you don't actually reference them later.

I do believe that the imports vs exports is a typo though.

- `B` - A package that imports `A`
- `Test::A` - A `test!` package, containing tests for the package `A`.

The cases where this error can show up in this set of packages are:

1. If `B` imports `A` with the `uses_internals: true` annotation, as only `test!` packages are allowed to import with `uses_internals: true`, or

> 2. if `B` imports `Test::A`, as only `test!` packages can import other packages marked `test!`.

Reviewer: ```suggestion
1. If `B` imports `A` with the `uses_internals: true` annotation, as only `test!` packages are allowed to import with `uses_internals: true`.
2. If `B` imports `Test::A`, as only `test!` packages can import other packages marked `test!`.
```

alternatively, leave the `, or` but make the `1.` item start with a lowercase `if`
